components/DateInput.vue

```diff
@@ -0,0 +1,169 @@
+        <input
+          :class="inputClass"
+          aria-describedby="yearinput"
+          :id="`${uniqueId}-${criteriaKey}-year`"
+          :name="`${uniqueId}-${criteriaKey}-year`"
+          type="text"
+          minlength="4"
+          maxlength="4"
+          pattern="[0-9]*"
+          inputmode="numeric"
+          :value="pullDateValue(dateResponse, 2)"
+          @change="updateElibilityDate($event, criteriaKey)"
+        />
+      </div>
+    </div>
+  </fieldset>
+</template>
+
+<style scoped>
+.usa-legend {
+  margin-bottom: -0.6875rem;
+}
+
+.usa-legend--empty,
+.usa-label--empty,
+.usa-input--empty {
+  color: #1b1b1b;
+}
+
+.usa-legend--success,
+.usa-label--success,
+.usa-input--success {
+  color: green;
+  font-weight: bold;
+}
+
+.usa-legend--error,
+.usa-label--error,
+.usa-input--error {
+  color: red;
+  font-weight: bold;
+}
+</style>
+
+<script>
+import _ from 'lodash'
+export default {
+    name: 'DateInput',
+    props: {
+        criteriaKey: {
+            type: String,
+            default: 'no criteria key provided'
+        },
+        label: {
+            type: String,
+            default: 'no label provided'
+        },
+        response: {
+            type: [String, Object, Boolean],
+            default: 'no response provided'
+        },
+        dateResponse: {
+          type: String,
+          default: 'no response inputted'
+        },
+        location: {
+          type: String,
+          default: "benefit-card",
+          validator: (value) => { return ['benefit-card', 'left-rail'].includes(value) }
+        }
+    },
+    data() {
+        return {
+            uniqueId: _.uniqueId('dateinput-')
+        }
+    },
+    computed: {
+      selectedStyle() {
+        if (this.location === "left-rail") {
+          return "text-bold"
+        }
+        return null
+      },
+      labelClass() {
+        return `usa-label usa-label--${this.response ? 'success': 
+            this.response == null ? 'empty': 'error'}`
+      },
+      legendClass() {
+        return `usa-legend usa-legend--${this.response ? 'success': this.response == null ? 'empty': 'error'} ${this.selectedStyle}`
+      },
+      inputClass() {
+        return `usa-input usa-input--${this.response ? 'success': 
+            this.response == null ? 'empty': 'error'}`
+      }
+    },
+    mounted() {
+        this.uniqueId = _.uniqueId('dateinput-')
+    },
+    methods: {
+        pullDateValue(dateResponse, index) {
+            return `${dateResponse !== null ? dateResponse.split('-')[index]: ''}`
+        },
+        updateElibilityDate (event, key) {
+            // figure out date from 3 boxes
+            const month = document.getElementById(`${this.uniqueId}-${this.criteriaKey}-month`).value
+            const day = document.getElementById(`${this.uniqueId}-${this.criteriaKey}-day`).value
+            const year = document.getElementById(`${this.uniqueId}-${this.criteriaKey}-year`).value
+            if (
```

Comment:
shouldn't need to do document.getElementById.   DOM should bind to data values in the component

Reply:
I am doing a document.getElementById to only use the singular method for all of the change events in the inputs for the D/M/Y